---
title: "" 
author: "`r rworkflows::use_badges()`" 
date: "<h4>Most recent update: <i>`r format( Sys.Date(), '%b-%d-%Y')`</i></h4>"
output:
  github_document 
editor_options: 
  chunk_output_type: inline
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
pkg <- read.dcf(file="DESCRIPTION",
                fields = "Package")[[1]]
```
 
## Intro

### About HPO

The [Human Phenotype Ontology (HPO)](https://hpo.jax.org/app/) controlled vocabulary of phenotypic abnormalities encountered in human disease. It currently contains over 18,000 hierarchically organised terms. Each term in the HPO describes a phenotypic abnormality, ranging from very broad phenotypes (e.g. "Abnormality of the nervous system") down to extremely specific phenotypes (e.g. "Decreased CSF 5-hydroxyindolacetic acid concentration"). 

The HPO is currently being used in thousands of exome and genome sequencing projects around the world to aid in the interpretation of human variation, in clinical practice to support differential diagnosis and to annotate patient information, and in research to understand the role of rare variants in human health and disease. The HPO was developed by the [Monarch Initiative](https://monarchinitiative.org/) in collaboration with [The Jackson Laboratory](https://www.jax.org/).

### About `HPOExplorer`

`HPOExplorer` is an R package with extensive functions for easily importing, annotating, filtering, and visualising the [Human Phenotype Ontology (HPO)](https://hpo.jax.org/app/) at the disease, phenotype, and gene levels. By pulling fresh data directly from official resources like [HPO](https://hpo.jax.org/app/), [Monarch](https://monarchinitiative.org/) and [GenCC](https://thegencc.org/), it ensures tightly controlled version coordination with the most up-to-date data available at any given time (with the option to use caching to boost speed). Furthermore, it can efficiently reorganise gene annotations into sparse matrices for usage within downstream statistical and machine learning analysis.

`HPOExplorer` was developed by the [Neurogenomics Lab](https://www.neurogenomics.co.uk/) at Imperial College London, along with valuable feedback provided by the HPO team. This package is still actively evolving and growing. Community engagement is welcome and any suggestions can be submitted as an [Issue](https://github.com/neurogenomics/HPOExplorer/issues) or [Pull Request](https://github.com/neurogenomics/HPOExplorer/pulls).

## Installation

Within R: 

```{r, eval=FALSE}
if (!require("remotes")) install.packages("remotes")
if (!require("HPOExplorer")) remotes::install_github("neurogenomics/HPOExplorer")

library(HPOExplorer)
```

## [Documentation website](https://neurogenomics.github.io/HPOExplorer/)

### [Getting started](https://neurogenomics.github.io/HPOExplorer/articles/HPOExplorer.html)

A quick tutorial on how to get started with `HPOExplorer`.

### [Docker](https://neurogenomics.github.io/HPOExplorer/articles/docker.html) 

`HPOExplorer` is also available via [DockerHub](https://hub.docker.com/repository/docker/neurogenomicslab/hpoexplorer). 
Click
[here](https://neurogenomics.github.io/HPOExplorer/articles/docker.html)
for instructions on how to create a Docker or Singularity
container with `HPOExplorer` and Rstudio pre-installed. 

# Citation

If you use ``r pkg``, please cite: 

<!-- Modify this by editing the file: inst/CITATION  -->
> `r utils::citation(pkg)$textVersion`

# Session Info 

<details> 

```{r Session Info}
utils::sessionInfo()
```

</details>  
<hr>
