<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Phenotypes to GenomicRanges — phenos_to_granges • HPOExplorer</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Phenotypes to GenomicRanges — phenos_to_granges"><meta property="og:description" content="Convert a HPO phenotype dataframe generated by
make_phenos_dataframe
to a GRangesList split by HPO ID.
The resulting object will contain genes (and gene metadata) for all
genes associated with each phenotypes."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">HPOExplorer</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/HPOExplorer.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/docker.html">Docker/Singularity Containers</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/neurogenomics/HPOExplorer/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Phenotypes to <a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GenomicRanges</a></h1>
    <small class="dont-index">Source: <a href="https://github.com/neurogenomics/HPOExplorer/blob/master/R/phenos_to_granges.R" class="external-link"><code>R/phenos_to_granges.R</code></a></small>
    <div class="hidden name"><code>phenos_to_granges.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Convert a HPO phenotype dataframe generated by
<a href="make_.html">make_phenos_dataframe</a>
to a <a href="https://rdrr.io/pkg/GenomicRanges/man/GRangesList-class.html" class="external-link">GRangesList</a> split by HPO ID.
The resulting object will contain genes (and gene metadata) for all
genes associated with each phenotypes.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">phenos_to_granges</span><span class="op">(</span></span>
<span>  phenos <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  phenotype_to_genes <span class="op">=</span> <span class="fu"><a href="load_phenotype_to_genes.html">load_phenotype_to_genes</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  hpo <span class="op">=</span> <span class="fu"><a href="get_.html">get_hpo</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  keep_chr <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">22</span><span class="op">)</span>, <span class="st">"X"</span>, <span class="st">"Y"</span><span class="op">)</span>,</span>
<span>  by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"hpo_id"</span>, <span class="st">"disease_id"</span><span class="op">)</span>,</span>
<span>  gene_col <span class="op">=</span> <span class="st">"intersection"</span>,</span>
<span>  split.field <span class="op">=</span> <span class="st">"hpo_id"</span>,</span>
<span>  as_datatable <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  allow.cartesian <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-phenos">phenos<a class="anchor" aria-label="anchor" href="#arg-phenos"></a></dt>
<dd><p>A data.table containing HPO IDs and other metadata.</p></dd>


<dt id="arg-phenotype-to-genes">phenotype_to_genes<a class="anchor" aria-label="anchor" href="#arg-phenotype-to-genes"></a></dt>
<dd><p>Output of
<a href="load_phenotype_to_genes.html">load_phenotype_to_genes</a> mapping phenotypes
to gene annotations.</p></dd>


<dt id="arg-hpo">hpo<a class="anchor" aria-label="anchor" href="#arg-hpo"></a></dt>
<dd><p>Human Phenotype Ontology object,
loaded from <a href="https://rdrr.io/pkg/KGExplorer/man/get_.html" class="external-link">get_ontology</a>.</p></dd>


<dt id="arg-keep-chr">keep_chr<a class="anchor" aria-label="anchor" href="#arg-keep-chr"></a></dt>
<dd><p>Chromosomes to keep.</p></dd>


<dt id="arg-by">by<a class="anchor" aria-label="anchor" href="#arg-by"></a></dt>
<dd><p>A vector of shared column names in <code>x</code> and <code>y</code> to merge on.
This defaults to the shared key columns between the two tables.
If <code>y</code> has no key columns, this defaults to the key of <code>x</code>.</p></dd>


<dt id="arg-gene-col">gene_col<a class="anchor" aria-label="anchor" href="#arg-gene-col"></a></dt>
<dd><p>Name of the gene column.</p></dd>


<dt id="arg-split-field">split.field<a class="anchor" aria-label="anchor" href="#arg-split-field"></a></dt>
<dd><p>A character string of a recognized column name in <code>df</code> that contains
      the grouping. This column defines how the rows of <code>df</code> are split and
      is typically a <code>factor</code> or <code>character</code> vector.  When
      <code>split.field</code> is not provided the <code>df</code> will be split by the
      number of rows.</p></dd>


<dt id="arg-as-datatable">as_datatable<a class="anchor" aria-label="anchor" href="#arg-as-datatable"></a></dt>
<dd><p>Return as a <a href="https://rdrr.io/pkg/data.table/man/data.table.html" class="external-link">data.table</a>.</p></dd>


<dt id="arg-allow-cartesian">allow.cartesian<a class="anchor" aria-label="anchor" href="#arg-allow-cartesian"></a></dt>
<dd><p>See <code>allow.cartesian</code> in <code><a href="https://rdrr.io/pkg/data.table/man/data.table.html" class="external-link">[.data.table</a></code>.</p></dd>


<dt id="arg-verbose">verbose<a class="anchor" aria-label="anchor" href="#arg-verbose"></a></dt>
<dd><p>Print messages.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A <a href="https://rdrr.io/pkg/GenomicRanges/man/GRangesList-class.html" class="external-link">GRangesList</a>.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">phenos</span> <span class="op">&lt;-</span> <span class="fu"><a href="make_.html">make_phenos_dataframe</a></span><span class="op">(</span>ancestor <span class="op">=</span> <span class="st">"Neurodevelopmental delay"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading cached RDS file: phenotype_to_genes.txt</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Version: 2025-01-16</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting data for 24 descendents.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Computing gene counts.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding term definitions.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding level-2 ancestor to each HPO ID.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding ancestor metadata.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Ancestor metadata already present. Use force_new=TRUE to overwrite.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 24 associations remain after filtering.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Getting absolute ontology level for 19,034 IDs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Computing ontology level / gene count ratio.</span>
<span class="r-in"><span><span class="va">grl</span> <span class="op">&lt;-</span> <span class="fu">phenos_to_granges</span><span class="op">(</span>phenos <span class="op">=</span> <span class="va">phenos</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting phenos to GRangesList.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading cached RDS file: phenotype_to_genes.txt</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Version: 2025-01-16</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding genes and disease IDs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required namespace: ensembldb</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gathering metadata for 2513 unique genes.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required namespace: EnsDb.Hsapiens.v75</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder, Robert Gordon-Smith, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer></div>






  </body></html>

